internal/compiler/parser/statements.rs

```diff
@@ -50,6 +55,11 @@ pub fn parse_statement(p: &mut impl Parser) -> bool {
         return true;
     }
 
+    if p.peek().as_str() == "let" {
```

Comment:
I know we have the same problem with `return`, but we should check that the next token is also an identifier. So if there is a property called `let`, code like `let = 3;` or `let + 1` still works.